framework/include/userobject/UserObjectInterface.h

```diff
@@ -84,14 +87,16 @@ template<class T>
 const T &
 UserObjectInterface::getUserObject(const std::string & name)
 {
-  return _uoi_feproblem.getUserObject<T>(_uoi_params.get<UserObjectName>(name));
+  unsigned int tid = isLocalUserObject(getUserObjectBase(name)) ? _uoi_tid : 0;
```

Comment:
I see, you mean keep a separate list of the Discrete* objects in the warehouse? This should probably be done eventually. We could get rid of the dynamic casts on get that way and we could guarantee that execute etc. are never called. 

But given that we have three people with separate PRs waiting on this functionality I'd like to pin down the user facing interface first and rework the warehouse stuff later. I don't even know it it makes sense to work on that part right now, with the warehouse refactor in progress.